js/src/logger.ts

```diff
@@ -489,17 +521,19 @@ function logFeedbackImpl(
     Object.entries(updateEvent).filter(([_, v]) => !isEmpty(v))
   );
 
-  const trueParentIds = new LazyValue(async () => {
-    const { kind, ...ids } = await parentIds.get();
-    return ids;
-  });
+  const parentIds = async () =>
+    new SpanParentComponents({
+      objectType: parentObjectType,
+      objectId: await parentObjectId.get(),
```

Comment:
Not sure what exactly you're referring to by "this id", but the `SpanParentComponents` object consists of three separate things: an object type enum (experiment or project logs), an object id (experiment id or project id), and a row id. I figured it was valid for people to serialize a reference to just the toplevel object without referring to a particular row if for instance, you want to distributed log a root span to some experiment/logger, so I permit passing an empty `rowId` which then gets serialized as `[object_type_prefix]:[object_id]:`.

Reply:
I see, but why are we generating that here? Is the default to serialize such a reference as the parent, unless you pass in a parent which includes a row id as well?

Reply:
I'm just using the `SpanParentComponents` object as a convenient way to generate parentIds regardless of whether it's used for spans or not. It contains logic for filling in `log_id="g"`, for instance. Maybe it makes sense to rename `SpanParentComponents` to something more general like `IdentifierComponents`?

Also I suppose `parentIds` as a variable name is somewhat of a misnomer because these are the object ids. There is no notion of parentage in logging feedback (I think?).

Reply:
Yeah makes sense. I think it's fine as is